### PR TITLE
CORE-1196: BRAvalancheTransaction

### PR DESCRIPTION
**Pull request by EBGToo**
Marked as a 'draft' as there is more to come; but this is expected to be the bulk of the work.  All 'prototype' tests pass; the fuji transaction of https://testnet.avascan.info/blockchain/x/tx/2T7WkhEYnypDQFGfB8LgNKkfPhVsmBsdyfPfy7fTm6XnaV2DXS has been reproduced.

### CODE REVIEW

File: WalletKitCore/src/avalanche/BRAvalancheNetwork.c [hunk 1]
```diff
@@ -50,14 +50,15 @@ avalancheNetworkGetAddressPrefix (BRAvalancheNetwork network) {
 
 #define AVALANCHE_NETWORK_STRIP_ADDRESS_LABEL(network, string)                        \
   do {                                                                                \
-    if (NULL != network->addressLabel && strcmp (network->addressLabel, string) <= 0) \
-      string = &string[strlen(network->addressLabel)];                                \
+    char *label = network->addressLabel;                                              \
+    if (NULL != network->addressLabel && 0 == strncmp (label, string, strlen(label))) \
```
Comment (bgoring) [inline WalletKitCore/src/avalanche/BRAvalancheNetwork.c:54]: Why not use the alias here in the NULL check?

File: WalletKitCore/src/avalanche/BRAvalancheTransaction.c [hunk 2]
```diff
@@ -2,22 +2,534 @@
...
+    array_free_all (addressEncodings, dataFree);
+
+    return result;
+}
+
+extern OwnershipGiven BRData
+avalancheTransactionOutputEncode (BRAvalancheTransactionOutput output) {
+    BRData assetEncoding     = avalancheHashEncode(output.assetHash);
+    BRData addressesEncoding = avalancheAddressXArrayEncode(output.addresses);
+    BRData typeEncodiing     = dataCreateEmpty();
```
Comment (bgoring) [inline WalletKitCore/src/avalanche/BRAvalancheTransaction.c:173]: typeEncoding

File: WalletKitCore/src/avalanche/BRAvalancheTransaction.c [hunk 3]
```diff
@@ -2,22 +2,534 @@
...
+
+static SortCompareRoutine
+avalancheUTXOSearchGetCompareRoutine (BRAvalancheUTXOSearchType type) {
+    static SortCompareRoutine routines[] = {
+        avalancheUTXOSortByAmountIncreasing,
+        avalanceUTXOSortByAmountDecreasing,
+        NULL,
+        avalancheUTXOSortByDateIncreasing
+    };
+    return routines[type];
```
Comment (bgoring) [inline WalletKitCore/src/avalanche/BRAvalancheTransaction.c:469]: `BRAvalancheUTXOSearchType`   has 3 enumerations?

File: WalletKitCore/src/avalanche/BRAvalancheTransaction.c [hunk 4]
```diff
@@ -2,22 +2,534 @@
...
+
+            if (*amountUTXOs >= amountWithFee) break /* from for-loop */;
+        }
+    }
+
+    // Check if `amount` was met
+    if (*amountUTXOs < amountWithFee) {
+        // Return the utxos in `result`
+        for (size_t index = 0; index < array_count(result); index++) {
+            if (updateUTXOs) BRSetAdd (utxos, result[index]);
```
Comment (bgoring) [inline WalletKitCore/src/avalanche/BRAvalancheTransaction.c:523]: Does the order of items in utxos change here by virtue of having taken out, and now being returned when `amountWithFee` is not met, and does it matter?

**Review by bgoring — APPROVED** [review]

File: WalletKitCore/src/avalanche/BRAvalancheNetwork.c (hunk 1 above)
Comment (EBGToo) [reply WalletKitCore/src/avalanche/BRAvalancheNetwork.c:54]: Fixed

File: WalletKitCore/src/avalanche/BRAvalancheTransaction.c (hunk 2 above)
Comment (EBGToo) [reply WalletKitCore/src/avalanche/BRAvalancheTransaction.c:173]: Fixed

File: WalletKitCore/src/avalanche/BRAvalancheTransaction.c (hunk 4 above)
Comment (EBGToo) [reply WalletKitCore/src/avalanche/BRAvalancheTransaction.c:523]: `utxos` is a BRSet; no ordering implied.

**Pull request merged**